### PR TITLE
refactor cast logs to use alloy

### PR DESCRIPTION
**Pull request by bernard-wagner**
Removed all the ethers-rs components used for `cast logs`. 

Use `resolve()` to convert arguments to DynSolType, replacing the previous `kind` property.

Previously could use `event.filter()` to convert indexed arguments into topic hashes, now explicitly calling abi-encode on provided arguments. Defaulting to an empty topic if the argument is empty.

Updated the unit tests and tried locally using my bash history for `cast logs`.

### CODE REVIEW

File: crates/cast/bin/cmd/logs.rs [hunk 1]
```diff
@@ -115,40 +110,29 @@ fn build_filter(
...
+    let mut filter = filter.select(block_option);
+
+    if let Some(address) = address {
+        filter = filter.address(address)
+    }
+
     Ok(filter)
 }
 
 /// Creates a TopicFilter from the given event signature and arguments.
```
Comment (Evalir) [inline crates/cast/bin/cmd/logs.rs:134]: Let's update the doc comments to be up to date on the file!

**Review by Evalir — COMMENTED** [review]
thanks a lot for this! would appreciate a bit of info on the PR description on:
- how you approached the mig
- if you tested this apart from running unit tests

File: crates/cast/bin/cmd/logs.rs [hunk 2]
```diff
@@ -115,40 +111,29 @@ fn build_filter(
...
+    if let Some(address) = address {
+        filter = filter.address(address)
+    }
+
     Ok(filter)
 }
 
-/// Creates a TopicFilter from the given event signature and arguments.
-fn build_filter_event_sig(event: Event, args: Vec<String>) -> Result<TopicFilter, eyre::Error> {
+/// Creates a Filter from the given event signature and arguments.
```
Comment (Evalir) [inline crates/cast/bin/cmd/logs.rs:135]:
```suggestion
/// Creates a [Filter] from the given event signature and arguments.
```

File: crates/cast/bin/cmd/logs.rs [hunk 3]
```diff
@@ -158,128 +143,74 @@ fn build_filter_event_sig(event: Event, args: Vec<String>) -> Result<TopicFilter
...
+        .topic3(topics[2].clone());
 
-    // Let filter do the hardwork of converting arguments to topics
-    Ok(event.filter(raw)?)
+    Ok(filter)
 }
 
-/// Creates a TopicFilter from raw topic hashes.
-fn build_filter_topics(topics: Vec<String>) -> Result<TopicFilter, eyre::Error> {
+/// Creates a Filter from raw topic hashes.
```
Comment (Evalir) [inline crates/cast/bin/cmd/logs.rs:186]:
```suggestion
/// Creates a [Filter] from raw topic hashes.
```

**Review by Evalir — APPROVED** [review]
This is looking clean! pending @mattsse

**Review by mattsse — APPROVED** [review]
short and sweet, tysm!

**Pull request merged**